modules/effects/src/effect_sources.ts

```diff
@@ -47,7 +41,21 @@ export class EffectSources extends Subject<any> {
   toActions(): Observable<Action> {
     return this.pipe(
       groupBy(getSourceForInstance),
-      mergeMap(source$ => source$.pipe(groupBy(effectsInstance))),
+      mergeMap(source$ => {
+        return source$.pipe(
+          groupBy(effectsInstance),
+          tap(() => {
+            if (
+              onInitEffects in source$.key &&
+              typeof source$.key[onInitEffects] === 'function'
+            ) {
+              this.store.dispatch(
+                source$.key[onInitEffects].bind(source$.key)()
```

Comment:
this would be the same as `((key: OnInitEffects) => key.ngrxOnInitEffects())(source$.key)`, right?

Do we need that much complexity here? maybe just `this.store.dispatch(source$.key.ngrxOnInitEffects())`? (after we get it through type-guard)

Reply:
Great catch, I did it to bind `this` to the effect instance inside the `isOnInitEffects` function of the effect.

But this won't be possible AFAIK because we're using `Object.getPrototypeOf` to group the effect instances.